src/main/python/ttconv/tt.py

```diff
@@ -37,36 +37,81 @@
-      fill        - Optional  : bar fill character (Str)
-      print_end    - Optional  : end character (e.g. "\r", "\r\n") (Str)
+  A handler class which allows the cursor to stay on
+  one line for selected messages
   """
-  percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))
-  filled_length = int(length * iteration // total)
-  bar_val = fill * filled_length + '-' * (length - filled_length)
-  print(f'\r{prefix} |{bar_val}| {percent}% {suffix}', end = print_end)
-  # Print New Line on Complete
-  if iteration == total: 
-    print()
+  is_writing_progress_bar = False
+  last_progress_msg = ""
+
+  def progress_str(self, percent_progress: float) -> str:
+    '''Formats the progress string.'''
+
+    total = 1.0
+    prefix = "Reading Progress:" if READING else "Writing Progress:"
+    suffix = 'Complete'
+    decimals = 1
+    length = 50
+    fill = '█'
+    percent = ("{0:." + str(decimals) + "f}").format(100 * (percent_progress / float(total)))
+    filled_length = int(length * percent_progress // total)
+    bar_val = fill * filled_length + '-' * (length - filled_length)
+    
+    return f'\r{prefix} |{bar_val}| {percent}% {suffix}'
+
+  def emit(self, record):
+    try:
+      msg = self.format(record)
+
+      stream = self.stream
+      writing_progress_bar = hasattr(record, 'writing_progress_bar')
+
+      percent_progress = None
+      if writing_progress_bar:
+        percent_progress = getattr(record, 'percent_progress')
+        msg = self.progress_str(float(percent_progress))
+        self.last_progress_msg = msg
+  
+      if self.is_writing_progress_bar and not writing_progress_bar:
+        # erase and over write the progress bar
+        stream.write('\r')
+        length = len(self.last_progress_msg)
+        stream.write(' ' * length)
+
+        # go to beginning of the line and write the new messages
+        stream.write('\r')
+        stream.write(msg)
+        stream.write(self.terminator)
+
+        # write the old progress information
+        stream.write(self.last_progress_msg)
+      elif writing_progress_bar:
+        stream.write(msg)
+      else:
+        stream.write(msg)
+        stream.write(self.terminator)
+
+      if writing_progress_bar:
+        self.is_writing_progress_bar = True
+        if percent_progress is not None and float(percent_progress) >= 1.0:
+          sys.stdout.write(str('\r\n'))
+          self.is_writing_progress_bar = False
+
+      self.flush()
+
+    except (KeyboardInterrupt, SystemExit):
+      raise
+    except:
+      self.handleError(record)
 
 def progress_callback(percent_progress: float):
   '''Callback handler used by reader and writer.'''
-  prefix_str = "Reading Progress:" if READING else "Writing Progress:"
-  print_progress_bar(percent_progress, 1.0, prefix = prefix_str, suffix = 'Complete', length = 50)
+  LOGGER.info('', extra={'writing_progress_bar':True, 'percent_progress':percent_progress})
```

Comment:
Suggest adding a log message even it is ignored by the handler.
```suggestion
  LOGGER.info(f'{percent_progress}%', extra={'writing_progress_bar':True, 'percent_progress':percent_progress})
```

Reply:
Fixed